Tasks/ServiceFabricDeploy/ServiceFabricSDK/Publish-UpgradedServiceFabricApplication.ps1

```diff
@@ -177,46 +177,45 @@ function Publish-UpgradedServiceFabricApplication
+    $oldApplication = Get-ServiceFabricApplication -ApplicationName $ApplicationName
 
-        if (!$oldApplication)
+    if (!$oldApplication)
+    {
+        $errMsg = (Get-VstsLocString -Key SFSDK_AppDoesNotExist -ArgumentList $ApplicationName)
+        throw $errMsg
+    }
+    else
+    {
+        if($oldApplication.ApplicationTypeName -ne $names.ApplicationTypeName)
         {
-            $errMsg = (Get-VstsLocString -Key SFSDK_AppDoesNotExist -ArgumentList $ApplicationName)
+            $errMsg = (Get-VstsLocString -Key SFSDK_AppTypeMismatch -ArgumentList $ApplicationName)
             throw $errMsg
         }
-        else
-        {
-            if($oldApplication.ApplicationTypeName -ne $names.ApplicationTypeName)
-            {
-                $errMsg = (Get-VstsLocString -Key SFSDK_AppTypeMismatch -ArgumentList $ApplicationName)
-                throw $errMsg
-            }
-
-            if($SkipUpgradeSameTypeAndVersion -And $oldApplication.ApplicationTypeVersion -eq $names.ApplicationTypeVersion)
-            {
-                Write-Warning (Get-VstsLocString -Key SFSDK_SkipUpgradeWarning -ArgumentList @($names.ApplicationTypeName, $names.ApplicationTypeVersion))
-                return
-            }
-        }
 
-        ## Check upgrade status
-        $upgradeStatus = Get-ServiceFabricApplicationUpgrade -ApplicationName $ApplicationName
-        if ($upgradeStatus.UpgradeState -ne "RollingBackCompleted" -and $upgradeStatus.UpgradeState -ne "RollingForwardCompleted")
+        if($SkipUpgradeSameTypeAndVersion -And $oldApplication.ApplicationTypeVersion -eq $names.ApplicationTypeVersion)
         {
-            $errMsg = (Get-VstsLocString -Key SFSDK_UpgradeInProgressError -ArgumentList $ApplicationName)
-            throw $errMsg
+            Write-Warning (Get-VstsLocString -Key SFSDK_SkipUpgradeWarning -ArgumentList @($names.ApplicationTypeName, $names.ApplicationTypeVersion))
+            return
         }
+    }
 
-        $reg = Get-ServiceFabricApplicationType -ApplicationTypeName $names.ApplicationTypeName | Where-Object  { $_.ApplicationTypeVersion -eq $names.ApplicationTypeVersion }
-        if ($reg)
-        {
-            Write-Host (Get-VstsLocString -Key SFSDK_UnregisteringExistingAppType -ArgumentList @($names.ApplicationTypeName, $names.ApplicationTypeVersion))
-            $reg | Unregister-ServiceFabricApplicationType -Force
-        }
+    ## Check upgrade status
+    $upgradeStatus = Get-ServiceFabricApplicationUpgrade -ApplicationName $ApplicationName
+    if ($upgradeStatus.UpgradeState -ne "RollingBackCompleted" -and $upgradeStatus.UpgradeState -ne "RollingForwardCompleted")
+    {
+        $errMsg = (Get-VstsLocString -Key SFSDK_UpgradeInProgressError -ArgumentList $ApplicationName)
+        throw $errMsg
+    }
 
+    $reg = Get-ServiceFabricApplicationType -ApplicationTypeName $names.ApplicationTypeName | Where-Object  { $_.ApplicationTypeVersion -eq $names.ApplicationTypeVersion }
+    if ($reg)
```

Comment:
Yes. Till this place, action will always be RegisterAndUpgrade since we've hardcoded this action in our deploy.js before invoking this file.

Reply:
I'd rather not start making those assumptions here as they'll bite us later if we open up more options in the UI and it will make it harder to merge the scripts.

Reply:
We are not giving any options in the UI its just that we internally are deciding weather we should RegisterAndUpgrade or just Upgrade.

Reply:
If you're not going to support the other actions then you must remove them from $Action's validation set and documentation. I'd prefer we stay closer to the SDK scripts and support them but we can't have the script document them as supported and then not do it. That is a maintainability problem waiting to happen.